lib/proxy/router.go

```diff
@@ -282,16 +285,31 @@ func (r *Router) DialHost(ctx context.Context, clientSrcAddr, clientDstAddr net.
 				}
 			}
 		}
-
 	} else {
+		// DELETE IN 15.0.0: necessary for smoothing over v13 to v14 transition only.
+		// In v15 the connection problem error should always be returned here.
 		if !r.permitUnlistedDialing {
 			return nil, trace.ConnectionProblem(errors.New("connection problem"), "direct dialing to nodes not found in inventory is not supported")
 		}
+
+		// Prepare a dummy server resource so this connection will not be
+		// treated like a connection to a Teleport node
+		isUnknownNode = true
+		isAgentlessNode = true
 		if port == "" || port == "0" {
 			port = strconv.Itoa(defaults.SSHServerListenPort)
 		}
-
 		serverAddr = net.JoinHostPort(host, port)
+		name := "unknown server " + serverAddr
+		target, err = types.NewServer(name, types.KindNode, types.ServerSpecV2{
+			Addr:     serverAddr,
+			Hostname: host,
+		})
+		if err != nil {
+			return nil, trace.Wrap(err, "error creating server resource")
```

Comment:
Nit: Not sure this message adds a lot of useful context.
```suggestion
			return nil, trace.Wrap(err)
```

Reply:
Done